File: easy/0504_base_7/Solution.java
import java.util.ArrayDeque;
import java.util.Deque;

class Solution {
  public String convertToBase7(int num) {
    if (Math.abs(num) < 6) {
      return Integer.toString(num);
    }

    final boolean isNegative = num < 0;
    if (isNegative) {
      num  = -num;
    }

    final Deque<String> dq = new ArrayDeque<>();

    while (num != 0) {
      dq.addFirst(Integer.toString(num % 7));
      num /= 7;
    }

    if (isNegative) {
      dq.addFirst("-");
    }

    return String.join("", dq);
  }
}

File: easy/0504_base_7/solution.cpp
#include <deque>
#include <string>

class Solution {
public:
  std::string convertToBase7(int num) {
    if (abs(num) < 6)
      return std::to_string(num);

    const bool isNegative = num < 0;
    if (isNegative)
      num = -num;

    std::deque<int> dq;

    while (num) {
      dq.push_front(num % 7);
      num /= 7;
    }

    std::string base7Num;

    if (isNegative)
      base7Num.push_back('-');

    for (const int &digit : dq)
      base7Num.push_back(48 + digit);

    return base7Num;
  }
};
